### PR TITLE
Move external network E2E to main E2E matrix, add to flake finder

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

File: .github/workflows/e2e-full.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         deploytool: ['operator']
+        external_net: ['']
```
Comment (dfarrell07) [inline .github/workflows/e2e-full.yml:18]: Before anyone asks, yes it is required to include this default `external_net: ['']`. I first sent exactly this patch without those lines and no external net E2E jobs were defined.

**Comment by dfarrell07** [comment]
This will require an update to the required jobs config in GH.

File: .github/workflows/e2e-full.yml [hunk 2]
```diff
@@ -23,35 +24,27 @@ jobs:
             globalnet: 'globalnet'
           - ovn: 'ovn'
             cable_driver: 'wireguard'
+        include:
+          - external_net: 'external-net'
+          - external_net: 'external-net'
+            globalnet: 'globalnet'
```
Comment (mkimuram) [inline .github/workflows/e2e-full.yml:30]: As I read https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs#example-including-new-combinations , I understand that only the combinations of ('external-net') and ('external-net', 'globalnet') is added by this include lines. And these are currently defined in e2e-external job, and are merged to this e2e job. Looks good to me.

**Comment by mkimuram** [comment]
Also confirmed that proper external network tests passed in "End to End Full". And the flake finder is defined in the same way, so it should work well (I guess that we need to merge to check if it actually works well in https://github.com/submariner-io/submariner/actions/workflows/flake_finder.yml).

- `End to End Full / E2E (external-net) (pull_request) Successful in 13m`
https://github.com/submariner-io/submariner/runs/5792701103?check_suite_focus=true

```
Ran 6 of 26 Specs in 72.015 seconds
2821
SUCCESS! -- 6 Passed | 0 Failed | 7 Pending | 13 Skipped
```

- `End to End Full / E2E (external-net, globalnet) (pull_request) Successful in 14m`
https://github.com/submariner-io/submariner/runs/5792701164?check_suite_focus=true

```
Ran 13 of 26 Specs in 259.893 seconds
3094
SUCCESS! -- 13 Passed | 0 Failed | 7 Pending | 6 Skipped
```

@dfarrell07 It seems that below job is still shown. Is it something that will be disappear once this PR is merged?

- `External Network E2E (external-net, globalnet) Expected — Waiting for status to be reported`

Once it becomes clear, I will approve. Anyway, thank you very much for working on this to improve!

File: .github/workflows/e2e-full.yml (hunk 1 above)
Comment (mkolesnik) [reply .github/workflows/e2e-full.yml:18]: This seems to be a quirk in GHA though on their documentation it seems like it should be working.

Perhaps open a bug for this and link to the two runs (or maybe such a bug is already open)?

**Review by mkolesnik — COMMENTED** [review]
> This will require an update to the required jobs config in GH.

I remember leaving this comment but can't find it (maybe another such patch?) so:

I think the best course of action is to place all experimental features (this, wireguard, etc) in a separate `e2e-experimental.yaml` or such file.
They could still trigger from `ready-to-test` label but it would be much easier to differentiate (especially for branch protections) which jobs are required for merging and which are experimental features (sometimes more flaky) which aren't mandatory for merging a patch.

**Comment by dfarrell07** [comment]
> I think the best course of action is to place all experimental features (this, wireguard, etc) in a separate e2e-experimental.yaml or such file.

I don't really want to get in to the business of defining what's stable and what's not by moving jobs between E2E workflows. I think that would result in lots more required jobs updates. Both External E2E and Wireguard were required jobs. It's not clear to me that Wireguard is more experimental than VXLAN. Let's leave discussions of stability for explicit docs, not kinda-arbitrary workflow splits.

That's also basically the inverse of the goal of this patch - to avoid bugs and reduce maintenance by de-duplicating. This adds a post-mortem and makes the timeout constant, but in the future the two workflows might diverge in different ways. If we go back to splitting the on-PR workflows, we'd also probably want to keep the on-timer workflows split for kinda-consistent maintenance. 

So if you want that change @mkolesnik you can send a PR and we can discuss, but that's not what I propose.

**Comment by dfarrell07** [comment]
> It seems that below job is still shown. Is it something that will be disappear once this PR is merged?

@mkimuram That's GitHub configuration for the jobs required to pass. Since this job was renamed it's still looking for the old job.
An Owner will need to update that configuration when this is merged.

**Comment by mkolesnik** [comment]
Fine by me to better discuss and define it, but we should put a pin in this PR until it's decided since this change will make external E2E required for any PR

**Comment by dfarrell07** [comment]
> this change will make external E2E required for any PR

I'm not sure I understand what you mean. As you can see, it's not currently required and that shouldn't change after merge. It will run on every PR yes, but I think that's what we want and currently have.

Maybe you're looking at the old job, which was required?

File: .github/workflows/e2e-full.yml [hunk 3]
```diff
@@ -15,6 +15,7 @@ jobs:
       fail-fast: false
```
Comment (sridhargaddam) [inline .github/workflows/e2e-full.yml:15]: Do we have to increase the timeout-minutes as we are now including this new external_net jobs?

**Comment by sridhargaddam** [comment]
> > This will require an update to the required jobs config in GH.
> 
> I remember leaving this comment but can't find it (maybe another such patch?) so:
> 
> I think the best course of action is to place all experimental features (this, wireguard, etc) in a separate `e2e-experimental.yaml` or such file. They could still trigger from `ready-to-test` label but it would be much easier to differentiate (especially for branch protections) which jobs are required for merging and which are experimental features (sometimes more flaky) which aren't mandatory for merging a patch.

I'm fine either way, as long as we get some feedback about stability of external_net jobs using the flake finder. Please note that for supporting external_network use-cases there are no explicit changes in Submariner. The changes are mostly outside of Submariner, but validating the use-case regularly will allow us to know early if the external-net use-case is broken due to internal changes in Submariner.

File: .github/workflows/e2e-full.yml (hunk 3 above)
Comment (dfarrell07) [reply .github/workflows/e2e-full.yml:15]: The External Net job was running with 30m timeout before, this changes it to use the same 45m timeout the other E2E jobs use.

**Comment by dfarrell07** [comment]
@mkolesnik Please re-review

**Review by mkolesnik — APPROVED** [review]

**Review by skitt — APPROVED** [review]

**Pull request merged**